lib/autoyast.pm

```diff
@@ -667,6 +668,56 @@ sub expand_version {
     return $profile;
 }
 
+=head2 expand_agama_profile
+
+ expand_agama_profile($profile);
+
+ Expand PRODUCT_ID, SCC_REGCODE etc.
+
+ $profile is the agama profile.
+
+=cut
+
+sub expand_agama_profile {
+    my ($profile, $profile_expanded) = @_;
+    $profile_expanded //= $profile;
+    my $content = expand_variables(get_test_data($profile));
+    save_tmp_file($profile_expanded, $content);
+    my $str = '';
+    my $product = get_var('SLE_PRODUCT');
+    if ($content =~ /PRODUCT_ID/) {
+        if ($product eq 'sles') {
+	    if (get_var('TEST') =~ 'sles4sap') {
+                $str = uc($product) . '_' . 'SAP' . '_' . '16.0';	
+	    }
+	    else {
+                $str = uc($product) . '_' . '16.0';
+            }
+        }
+        if (get_var('DISTRI') eq 'opensuse') {
+            if (get_var('LEAP')) {
+                $str = 'Leap' . '_' . get_var('VERSION');
+	    }
+	    else {
+                $str = 'Tumbleweed';
+	    }
+        }
+        $content =~ s/\{\{PRODUCT_ID\}\}/$str/g;
+    }
+    if ($content =~ /SCC_REGCODE/) {
+        if ($product eq 'sles') {
+            if (get_var('TEST') =~ 'sles4sap') {
+                $str = get_var('SCC_REGCODE_SLES4SAP');
+	    }
+	    else {
+                $str = get_var('SCC_REGCODE');
+            }
+        }
+	$content =~ s/\{\{SCC_REGCODE\}\}/$str/g;
+    }
+    upload_profile(path => $profile_expanded, profile => $content);
+}
+
```

Comment:
I found this code an overkill in general, hard to maintain compare with solution of having more jsonnet files, besides we cannot support our test in a variable like TEST. I need to think about this...